rust/src/delta.rs

```diff
@@ -950,25 +950,30 @@ impl DeltaTable {
         note = "Please use the get_file_uris function instead"
     )]
     pub fn get_file_paths(&self) -> Vec<String> {
-        self.get_file_uris()
+        self.get_file_uris().collect()
     }
 
     /// Returns a URIs for all active files present in the current table version.
-    pub fn get_file_uris(&self) -> Vec<String> {
+    pub fn get_file_uris(&self) -> impl Iterator<Item = String> + '_ {
```

Comment:
I'd prefer for these to be iterator and saw similar changes in #452, so thought this would be alright. Happy to change it back if you don't want to change the public API in this way.

Reply:
this is a good change, we should propagate this practice through out the code base :)